Lib/test/test_pprint.py

```diff
@@ -447,6 +479,28 @@ def test_mapping_proxy(self):
                           ('lazy', 7),
                           ('dog', 8)]))""")
 
+    def test_dict_views(self):
+        for dict_class in (dict, collections.OrderedDict):
+            empty = dict_class()
+            short = dict_class(zip('abcde', 'abcde'))
+            long = dict_class((chr(x), chr(x)) for x in range(65, 91))
+            prefix = "dict" if dict_class is dict else "odict"
+            for d in empty, short, long:
+                is_short = len(d) < 6
+                joiner = ", " if is_short else ",\n "
+                k = d.keys()
+                v = d.values()
+                i = d.items()
+                self.assertEqual(pprint.pformat(k, sort_dicts=True),
+                                 prefix + "_keys([%s])" %
+                                 joiner.join(repr(key) for key in sorted(k)))
+                self.assertEqual(pprint.pformat(v, sort_dicts=True),
+                                 prefix + "_values([%s])" %
+                                 joiner.join(repr(val) for val in sorted(v)))
+                self.assertEqual(pprint.pformat(i, sort_dicts=True),
+                                 prefix + "_items([%s])" %
+                                 joiner.join(repr(item) for item in sorted(i)))
```

Comment:
All tests sort keys, so the unsorted path is not covered?

Reply:
Added unsorted paths to tests. Figured out that because the test dicts were pre-sorted alphabetically, I missed that short views were unsortable (because they went through `_safe_repr` as generic objects). Currently have minimal support for sorting committed, but will need more time to flesh out whether it needs to handle recursive views. Thank you for bringing attention to what would be a pseudo test hiding unwanted behavior.